### ISSUE TITLE
Language parameter

### ISSUE THREAD

**Issue by hifabienne** [zitadel/zitadel#6251]
I am able to send a language parameter to the ZITADEL UIs. If a language parameter is sent, ZITADEL UI will always be shown in the set language.

**Acceptance Criteria**
- [ ] ZITADEL Console accepts a language parameter, and renders the UI with the specified language
- [ ] ZITADEL Login accepts a language parameter and renders the UI with the specified language

**Comment by ahmednfwela** [comment]
wouldn't it make sense to have a language selection box instead ? so users aren't forced to a specific language, and can select form all the languages

**Comment by hifabienne** [comment]
> wouldn't it make sense to have a language selection box instead ? so users aren't forced to a specific language, and can select form all the languages

Yes, you are right that would probably make sense. At least as an addition

**Comment by ARSHIL1804** [comment]
can I pick this up? If yes, are we going with this suggestion? https://github.com/zitadel/zitadel/issues/6251#issuecomment-1646562164

**Comment by fforootd** [comment]
> can I pick this up? If yes, are we going with this suggestion? [#6251 (comment)](https://github.com/zitadel/zitadel/issues/6251#issuecomment-1646562164)

Sure, lets us know if you hit a blocker, the current login code can be a little fuzzy 😁

**Comment by ARSHIL1804** [comment]
What should be the position of the language box and also you want the language parameter? @fforootd

**Comment by ahmednfwela** [comment]
> What should be the position of the language box and also you want the language parameter? @fforootd 
> 

personally I think it should be visible all the time, and at the top, with an icon next to it for accessibility.

**Comment by livio-a** [comment]
just noticed the issue and wanted to give some insights for the login UI:

we used to have a language selection in the registration forms:
https://github.com/zitadel/zitadel/commit/f40931e81ecb5ac0dcb8d5ab4bd6bc606de5578f

as we removed that, there's currently the possibility to specify the language (of the user to be created) by a query parameter (is set into a hidden form field). It will however not change the languange used to display the labels / texts in the UI

**Comment by elinashoko** [comment]
Thanks for filing this issue. As part of an ongoing backlog cleanup, we're closing issues that have had no activity in over 12 months.

If this is still relevant, let us know in a comment (or reopen the issue) and we'll take another look.

This issue will close in 30 days if there's no further activity.
